clients/drcachesim/tools/filter/encodings2regdeps.h

```diff
@@ -0,0 +1,192 @@
+                       trace_encoding.size);
+                encoding_offset += trace_encoding.size;
+            }
+
+            /* Genenerate the real ISA instr_t by decoding the encoding bytes.
+             */
+            instr_t instr;
+            instr_init(dcontext_.dcontext, &instr);
+            app_pc next_pc = decode_from_copy(dcontext_.dcontext, encoding, pc, &instr);
+            if (next_pc == NULL || !instr_valid(&instr)) {
+                instr_free(dcontext_.dcontext, &instr);
+                error_string_ =
+                    "Failed to decode instruction " + to_hex_string(entry.addr);
+                return false;
+            }
+
+            /* Convert the real ISA instr_t into a regdeps ISA instr_t.
+             */
+            instr_t instr_regdeps;
+            instr_init(dcontext_.dcontext, &instr_regdeps);
+            instr_convert_to_isa_regdeps(dcontext_.dcontext, &instr, &instr_regdeps);
+            instr_free(dcontext_.dcontext, &instr);
+
+            /* Obtain regdeps ISA instr_t encoding bytes.
+             */
+            byte ALIGN_VAR(REGDEPS_ALIGN_BYTES)
+                encoding_regdeps[REGDEPS_MAX_ENCODING_LENGTH];
+            app_pc next_pc_regdeps =
+                instr_encode(dcontext_.dcontext, &instr_regdeps, encoding_regdeps);
+            instr_free(dcontext_.dcontext, &instr_regdeps);
+            if (next_pc_regdeps == NULL) {
+                error_string_ =
+                    "Failed to encode regdeps instruction " + to_hex_string(entry.addr);
+                return false;
+            }
+
+            /* Compute number of trace_entry_t to contain regdeps ISA encoding.
+             * Each trace_entry_t record can contain 8 byte encoding.
+             */
+            uint trace_entry_encoding_size = (uint)sizeof(entry.addr); /* == 8 */
+            uint regdeps_encoding_size = (uint)(next_pc_regdeps - encoding_regdeps);
+            uint num_regdeps_encoding_entries =
+                ALIGN_FORWARD(regdeps_encoding_size, trace_entry_encoding_size) /
+                trace_entry_encoding_size;
+            last_encoding.resize(num_regdeps_encoding_entries);
+
+            /* Copy regdeps ISA encoding, splitting it among the last_encoding
+             * trace_entry_t records.
+             */
+            uint regdeps_encoding_offset = 0;
+            for (trace_entry_t &encoding_entry : last_encoding) {
+                encoding_entry.type = TRACE_TYPE_ENCODING;
+                uint size = regdeps_encoding_size < trace_entry_encoding_size
+                    ? regdeps_encoding_size
+                    : trace_entry_encoding_size;
+                encoding_entry.size = (unsigned short)size;
+                memset(encoding_entry.encoding, 0, trace_entry_encoding_size);
+                memcpy(encoding_entry.encoding,
+                       encoding_regdeps + regdeps_encoding_offset, encoding_entry.size);
+                regdeps_encoding_size -= trace_entry_encoding_size;
```

Comment:
encoding_entry.size should be subtracted

Reply:
Right, thanks. Fixed.